dialvia/http_test.go

```diff
@@ -134,6 +135,58 @@ func TestHTTPProxyDialerDialContext(t *testing.T) {
 	})
 }
 
+func TestHTTPProxyConnectRequestModifier(t *testing.T) {
```

Comment:
I love tests but I guess this one adds no value tbh.

Reply:
OK I can remove it, since the implementation is short and straightforward. Moreover, there is another test of that in martian.